test/array.specs.ts

```diff
@@ -0,0 +1,81 @@
+import "mocha";
+import fc from "fast-check";
+import { chunk } from "../src/array"; 
+
+describe("chunk: ", () => {
+    it("Should split arrays into chunks of the specified size", () =>
+      fc.assert(
+        fc.property(
+          fc.array(fc.integer(), { minLength: 1 }), //tableau non vide
+          fc.nat(10), // la taille de chunk entre 0 et 10
+          (array, size) => {
+            const result = chunk(array, Math.max(size, 1)); // garantir que la taille du chunk est au moins 1
+            return result.length > 0 && result.every((chunk) => chunk.length <= Math.max(size, 1));
+          }
+        )
+      ));
+
+  it("Should return the original array when size is less than or equal to 0", () =>
+    fc.assert(
+      fc.property(fc.array(fc.integer()), fc.integer({ max: 0 }), (array, size) => {
+        const result = chunk(array, size);
+        return result.length === 1 && result[0] === array;
```

Comment:
That first === is ok, but you cannot compare arrays using === as you are comparing that they are the very same instance of array, while the function only guarantees that the resulting one chunk will contains the same elements, not the that it will be the same array.

Reply:
I removed the === now I ensure that the elements are identical even if they're not from the same array instance